includes/gateway/class-omise-payment-fbbot.php

```diff
@@ -0,0 +1,224 @@
+	require_once dirname( __FILE__ ) . '/class-omise-payment.php';
+
+	if ( ! class_exists( 'WC_Payment_Gateway' ) ) {
+		return;
+	}
+
+	if ( class_exists( 'Omise_Payment_FBBot' ) ) {
+		return;
+	}
+
+	class Omise_Payment_FBBot extends Omise_Payment {
+		private static $instance;
+		
+		public function __construct() {
+			parent::__construct();
+
+			$this->payment_page_url = "pay-on-messenger";
+			$this->payment_purchase_complete_url = "complete-payment";
+			$this->payment_error_url = "pay-on-messenger-error";
+			$this->omise_3ds      = $this->get_option( 'omise_3ds', false ) == 'yes';
+
+			add_action( 'wp_enqueue_scripts', array( $this, 'omise_assets' ) );
+
+			add_filter( 'the_posts', array( $this, 'payment_page_detect' ) );
+			add_filter( 'query_vars', array( $this, 'parameter_queryvars' ) );
+		}
+
+		public function omise_assets() {
+			wp_enqueue_style( 'omise-css', plugins_url( '../../assets/css/omise-css.css', __FILE__ ), array(), OMISE_WOOCOMMERCE_PLUGIN_VERSION );
+
+			wp_enqueue_script( 'omise-js', 'https://cdn.omise.co/omise.js', array( 'jquery' ), OMISE_WOOCOMMERCE_PLUGIN_VERSION, true );
+
+			wp_enqueue_script( 'omise-util', plugins_url( '../../assets/javascripts/omise-util.js', __FILE__ ), array( 'omise-js' ), OMISE_WOOCOMMERCE_PLUGIN_VERSION, true );
+
+      		wp_enqueue_script( 'omise-payment-on-messenger-form-handler', plugins_url( '../../assets/javascripts/omise-payment-on-messenger-form-handler.js', __FILE__ ), array( 'omise-js', 'omise-util' ), OMISE_WOOCOMMERCE_PLUGIN_VERSION, true );
+
+      		wp_localize_script( 'omise-payment-on-messenger-form-handler', 'omise_params', array(
+    			'key'       => $this->public_key()
+      		 ) );
+		}
+
+		public static function get_instance() {
+			if ( ! self::$instance ) {
+				self::$instance = new self();
+			}
+
+			return self::$instance;
+		}
+
+		public function parameter_queryvars( $qvars ) {
+			$qvars[] = 'product_id';
+			$qvars[] = 'messenger_id';
+    		$qvars[] = 'error_message';
+
+			return $qvars;
+		}
+
+		public function payment_page_detect( $posts ) {
+			global $wp;
+	    	global $wp_query;
+
+		    if ( $this->is_omise_purchase_complete_page() ) {
+		      	$image_url = site_url() . '/wp-content/plugins/omise-woocommerce/assets/images/omise_logo.png';
+		      	$url = 'https://www.messenger.com/closeWindow/?image_url=' . $image_url . '&display_text=THANKS%20FOR%20PURCHASE';
+		      	if ( wp_redirect( $url ) ) {
+		        	exit;
+		      	}
+
+		      	return $posts;
+		    }
+
+		    // Create custom page
+		    $post = new stdClass;
+
+		    if ( $this->is_omise_payment_page() ) {
+		    	$post->post_title = __( 'Your order' );
+		    	$post->post_content = $this->payment_page_render();
+
+		    } else if ( $this->is_omise_payment_error_page() ) {
+		      	$post->post_title = __( 'System error' );
+		      	$post->post_content = $this->payment_error_page_render();
+		    }
+
+		    if ( $this->is_accessible() ) {
+				// Create cumtom page content
```

Comment:
@igroomgrim thought you wanna go with this way, no? (Im ok both, just asking)
<img width="674" alt="screen shot 2560-08-01 at 10 12 34 pm" src="https://user-images.githubusercontent.com/2154669/28870171-7a21ee70-77a9-11e7-9423-c151ce305aff.png">

<img width="551" alt="screen shot 2560-08-02 at 5 40 19 pm" src="https://user-images.githubusercontent.com/2154669/28870235-b3944fb8-77a9-11e7-9ee3-9e4903244c34.png">

<img width="820" alt="screen shot 2560-08-01 at 10 14 31 pm" src="https://user-images.githubusercontent.com/2154669/28870187-8c05c38c-77a9-11e7-9a83-7119a1e89b53.png">